### DISCUSSION TITLE
Should BINDINGS use the appropriate character in Footer widget instead of the text description 'key' value?

### DISCUSSION THREAD

**Discussion by jspv** [Textualize/textual#1119] [Ideas]
Noting this change in behavior from 0.1.  When creating bindings for symbols like '=', '.', you now  use the name of the binding `equals_sign`, `full_stop` etc. for the bindings to be recognized.  Similar to:

```python
BINDINGS = [(
            "equals_sign",
            "load_menu",
            "Main Menu",
)]
```

This works, but has the effect of in the Footer widget, instead of seeing `=   Main Menu` in the Footer, I see `EQUALS_SIGN  Main Menu`.   I get around this now by using a full Binding class similar and passing `key_display` similar to:

```python
        Binding(
            key="full_stop", action="main_menu", description="Menu", key_display="."
        )
```
Shouldn't the character representation and not the full text representation be the default for the Footer?  Is there a use case for showing the text description rather than the key character?

**Comment by davep** [comment]
Currently the `Footer` class does take a very simplistic approach to this and it's something we're going to change. The idea will be that, where possible, the `Footer` will select a good `key_display` default for a given key if one isn't provided (or perhaps more likely a binding will pick a good default `key_display` if one isn't provided).

**Reply by davep** [reply]
Issue created for this: #1141
